internal/adapters/api/handler/role_permissions.go

```diff
@@ -0,0 +1,55 @@
+package handler
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/iBoBoTi/project_boiler_plate/internal/adapters/api/response"
+	"github.com/iBoBoTi/project_boiler_plate/internal/core/domain"
+	"github.com/iBoBoTi/project_boiler_plate/internal/core/ports"
+	"net/http"
+)
+
+type rolePermissionsHandler struct {
+	rolePermissionsService ports.RolePermissionsService
+	log                    ports.Logger
+}
+
+func NewRolePermissionsHandler(rolePermissionsService ports.RolePermissionsService, log ports.Logger) ports.RolePermissionsHandler {
+	return &rolePermissionsHandler{
+		rolePermissionsService: rolePermissionsService,
+		log:                    log,
+	}
+}
+
+func (h *rolePermissionsHandler) AddPermissionsToRole(c *gin.Context) {
+	rolePermissions := domain.RolePermissions{}
+	if err := c.ShouldBindJSON(&rolePermissions); err != nil {
+		response.JSON(c, "invalid_request_body", http.StatusBadRequest, nil, []string{err.Error()})
+		return
+	}
+	if err := h.rolePermissionsService.AddPermissionsToRole(&rolePermissions); err != nil {
+		response.JSON(c, "invalid_input", http.StatusBadRequest, nil, []string{err.Error()})
+		return
+	}
+	response.JSON(c, "permissions added successfully", http.StatusOK, nil, nil)
+}
+
+func (h *rolePermissionsHandler) RemovePermissionFromRole(c *gin.Context) {
+	roleID := c.Param("role_id")
+	permissionID := c.Param("permission_id")
```

Comment:
it will be nice if you are using ids like uuid that provides some sort of validation

Reply:
Done